devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy

```diff
@@ -0,0 +1,674 @@
+	} catch (e) {
+		log.error "$e"
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapCmd = cmd.encapsulatedCommand(commandClassVersions)
+
+	def result = []
+	if (encapCmd) {
+		result += zwaveEvent(encapCmd)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd) {
+	logDebug "Device Woke Up"
+
+	def cmds = []
+	if (state.refreshConfig || pendingChanges > 0) {
+		cmds += getConfigCmds()
+	}
+
+	if (canReportBattery()) {
+		cmds << batteryGetCmd()
+	}
+
+	if (state.refreshSensors) {
+		cmds += [
+				sensorBinaryGetCmd(),
+				sensorMultilevelGetCmd(tempSensorType),
+				sensorMultilevelGetCmd(lightSensorType)
+		]
+		state.refreshSensors = false
+	}
+
+	if (cmds) {
+		cmds = delayBetween(cmds, 1000)
+		cmds << "delay 3000"
+	}
+	cmds << secureCmd(zwave.wakeUpV1.wakeUpNoMoreInformation())
+	return response(cmds)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def val = (cmd.batteryLevel == 0xFF ? 1 : cmd.batteryLevel)
+	if (val > 100) {
+		val = 100
+	} else if (val < 1) {
+		val = 1
+	}
+	state.lastBatteryReport = new Date().time
+	state.lastBattery = val
+	logDebug "Battery ${val}%"
+	sendEvent(getEventMap("battery", val, null, null, "%"))
+	return []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	logTrace "SensorMultilevelReport: ${cmd}"
+
+	if (cmd.sensorValue != [255, 255]) { // Bug in beta device
+		switch (cmd.sensorType) {
+			case tempSensorType:
+				def unit = cmd.scale ? "F" : "C"
+				def temp = convertTemperatureIfNeeded(cmd.scaledSensorValue, unit, cmd.precision)
+
+				sendEvent(getEventMap("temperature", temp, true, null, getTemperatureScale()))
+				break
+
+			case lightSensorType:
+				sendEvent(getEventMap( "humidity", cmd.scaledSensorValue, true, null, "%"))
+				break
+			default:
+				logDebug "Unknown Sensor Type: ${cmd.sensorType}"
+		}
+	}
+	return []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	logTrace "ConfigurationReport ${cmd}"
+
+	runIn(4, refreshPendingChanges)
+
+	def param = configParams.find { it.num == cmd.parameterNumber }
+	if (param) {
+		def val = cmd.scaledConfigurationValue
+
+		logDebug "${param.name}(#${param.num}) = ${val}"
+		state["configParam${param.num}"] = val
+	} else {
+		logDebug "Parameter #${cmd.parameterNumber} = ${cmd.configurationValue}"
+	}
+	return []
+}
+
+def refreshPendingChanges() {
+	sendEvent(name: "pendingChanges", value: "${pendingChanges} Pending Changes", displayed: false)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	logTrace "NotificationReport: $cmd"
+	def result = []
+
+	if (cmd.notificationType == 0x06 && cmd.event == 0x16) {
```

Comment:
Could You use constants for notification types and events ?

Reply:
I can’t find information about official constant definitions such as "notificationv3.NotificationReport.NOTIFICATION_TYPE_BURGLAR"，So I have defined some constants according to the purpose.